### PR TITLE
fix(cloud-agent-next): log unclassified delivery error detail

### PR DESCRIPTION
**Pull request by eshurakov**
## Summary

`deliveryErrorLogFields` in `services/cloud-agent-next/src/sandbox-session/control-dispatch.ts` is the single projection behind the two delivery-failure warn logs in `SandboxSession` ("Control-plane dispatch failed" and "Queued prompt reconciliation failed"). It emitted `errorMessage` only for a `ControlRequestError`, so any unclassified control-plane transport or internal error reached telemetry with no detail at all, and the failure could not be diagnosed from logs.

Every delivery error now records its message, and the extraction is total: the helper runs inside catch blocks whose following statements record the failure, so a throwing string conversion falls back to `'[unserializable error]'` rather than propagating and silently skipping the failure recording.

Deliberately unchanged: `errorCode` still allowlists against `controlErrorCodes` and never copies an arbitrary peer-supplied code, and `retryable` still comes from `isRetryableDeliveryError`. No length bound and no secret redaction were added — `controlErrorSchema.message` is already unbounded on the classified branch, so bounding one half of the same field would be an unowned policy.

## Verification

Manually ran the local fake-LLM harness lifecycle `external-kill` (`pnpm --filter cloud-agent-next exec tsx test/e2e/run.ts external-kill echo:hi`) against the running local worker and stack to force a real control-plane delivery failure. The worker then logged the detail that was previously dropped:

- message `Control-plane dispatch failed`, phase `preparing`, errorCode `transport_or_internal_error`, errorMessage `Sandbox acquisition no longer owns this allocation`, retryable `false`.

The thrown error was a `SandboxAcquisitionLostError` (an ordinary `Error`, not a `ControlRequestError`), so this exercised exactly the branch the change alters. Before the change, that `errorMessage` field was absent for this record.

- [x] Manual harness run (`external-kill`) reproduced an unclassified delivery failure and showed the new field

Not manually exercised, flagged for transparency: the second call site ("Queued prompt reconciliation failed") did not fire in this run, and the `[unserializable error]` fallback is covered by unit tests only.

## Visual Changes

N/A

## Reviewer Notes

- The real edge case is the guard's placement. `deliveryErrorLogFields` is called at `SandboxSession.ts:4093` immediately before `recordDeliveryFailure`, so a throw inside the projection would skip the terminal accounting for a queued message entirely. The guard covers only the message conversion; the pre-existing throwing-Proxy hazards in `isRetryableDeliveryError` (`'retryable' in error`) are untouched and out of scope here.
- Added cases: an unclassified `Error`, a non-`Error` thrown value, `Object.create(null)`, and an object whose `toString` throws. The existing cases still pin that spoofable own properties other than the message (`code`, `stack`, `cause`, `overloaded`, `auth`, `env`) are not copied into the log fields.

### CODE REVIEW

File: services/cloud-agent-next/src/sandbox-session/control-dispatch.ts [hunk 1]
```diff
@@ -98,12 +98,20 @@ export function isRetryableDeliveryError(error: unknown): boolean {
 }
 
 export function deliveryErrorLogFields(error: unknown) {
+  // A logging helper runs inside catch blocks, so a throwing conversion must
+  // not propagate and skip the recovery that follows the log call.
+  let errorMessage: string;
+  try {
+    errorMessage = error instanceof Error ? error.message : String(error);
```
[the remark this answers is not in the document]
Comment (eshurakov) [reply services/cloud-agent-next/src/sandbox-session/control-dispatch.ts]: Fixed in 68733e2834.

`deliveryErrorLogFields` now reads a non-`Error` value's own `message` when it is a string, and keeps `String(error)` otherwise. Only own properties are read (`Object.hasOwn`), matching the own-field discipline `reconstructControlRequestError` already uses when it builds its projection. The read stays inside the existing `try`, so an exotic own-message lookup still falls back to `'[unserializable error]'` instead of throwing and skipping the failure recording that follows the log call.

Added cases for the passed-through malformed rejection (`{ code: '', message: 'Invalid rejection', retryable: true }`), an inherited `message`, a non-string `message`, and a throwing `message` getter.

Coverage note: no harness lifecycle or fake-LLM directive can induce a malformed control-plane rejection, so this branch is covered by unit tests only. The end-to-end run re-confirmed the surrounding warn-log path, not this branch.

**Review by pandemicsyn — APPROVED** [review]
